src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/AbstractReferenceFinder.cs

```diff
@@ -236,6 +317,241 @@ protected static async Task<ImmutableArray<SyntaxToken>> GetIdentifierOrGlobalNa
+                if (syntaxFacts.GetNameForAttributeArgument(attributeArgument) != "Target")
+                {
+                    return false;
+                }
+
+                var attributeNode = attributeArgument!.Parent?.Parent;
+                if (attributeNode == null || !syntaxFacts.IsGlobalAttribute(attributeNode))
+                {
+                    return false;
+                }
+
+                // Check the attribute type matches 'SuppressMessageAttribute'.
+                var attributeSymbol = semanticModel.GetSymbolInfo(attributeNode, cancellationToken).Symbol?.ContainingType;
+                return suppressMessageAttribute.Equals(attributeSymbol);
+            }
+
+            static ReferenceLocation CreateReferenceLocation(
+                int offsetOfReferenceInToken,
+                SyntaxToken token,
+                SyntaxNode root,
+                Document document,
+                ISyntaxFacts syntaxFacts)
+            {
+                // We found a valid reference to the symbol in documentation comment ID string literal.
+                // Compute the reference span within this string literal for the identifier.
+                // For example, consider the suppression below for field 'Field' defined in type 'C':
+                //      [assembly: SuppressMessage("RuleCategory", "RuleId', Scope = "member", Target = "F:C.Field")]
+                // We compute the span for 'Field' within the target string literal.
+                // NOTE: '#' is also a valid char in documentation comment ID. For example, '#ctor' and '#cctor'.
+
+                var positionOfReferenceInTree = token.SpanStart + offsetOfReferenceInToken + 1;
+                var valueText = token.ValueText;
+                var length = 0;
+                while (offsetOfReferenceInToken < valueText.Length)
+                {
+                    var ch = valueText[offsetOfReferenceInToken++];
+                    if (ch == '#' || syntaxFacts.IsIdentifierPartCharacter(ch))
+                        length++;
+                    else
+                        break;
+                }
+
+                // We create a reference location of the identifier span within this string literal
+                // that represents the symbol reference.
+                // We also add the location for the containing documentation comment ID string literal.
+                // For the suppression example above, location points to the span of 'Field' inside "F:C.Field"
+                // and containing string location points to the span of the entire string literal "F:C.Field".
+                var location = Location.Create(root.SyntaxTree, new TextSpan(positionOfReferenceInTree, length));
+                var containingStringLocation = token.GetLocation();
+                return new ReferenceLocation(document, location, containingStringLocation);
+            }
+        }
+
+        private static bool GetExpectedDocumentationCommentId(
```

Comment:
This helper method and the one below would ideally be static local functions for the prior method, but I couldn't add nullable attributes to parameters if I did so.